src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/INodeWithDebugInfo.cs

```diff
@@ -19,12 +23,37 @@ public DebugLocInfo(int nativeOffset, string fileName, int lineNumber, int colNu
             ColNumber = colNumber;
         }
     }
-
+    
     public interface INodeWithDebugInfo
     {
         DebugLocInfo[] DebugLocInfos
         {
             get;
         }
+
+        DebugVarInfo[] DebugVarInfos
+        {
+            get;
+        }
+    }
+
+     
+    public struct DebugVarInfo
+    {
+        public readonly uint VarNumber; // For parameters, this is the parameter index.
+                                        // For local variables, this is the slot number.
+        public readonly string Name;
+        public readonly uint TypeIndex;
+        public readonly bool IsParam;
+        public List<NativeVarInfo> Ranges;
```

Comment:
If I was doing this, I would have a `DebugVarInfoBuilder` type (which is a mutable things and you can add new ranges to it), with a `ToDebugVarInfo` method that creates a thing where `Ranges` is immutable and you can't add new things to it anymore.

I generally prefer immutable things to transfer data within the compiler because with immutable structures you don't need to ever worry about multithreading.